File: tournaments/caesarBoxCipherEncoding/caesarBoxCipherEncoding.js
function caesarBoxCipherEncoding(inputString) {
    const n = Math.sqrt(inputString.length);
    let result = '';

    for (let i = 0; n > i; ++i)
        for (let j = 0; n > j; ++j)
            result += inputString[i + j * n];

    return result;
}

File: tournaments/caesarBoxCipherEncoding/caesarBoxCipherEncoding.cpp
std::string caesarBoxCipherEncoding(std::string inputString) {

  int n = std::sqrt(inputString.size());
  std::string ans = "";
  for (int i = 0; i < n; i++) {
    for (int j = 0; j < n; j++) {
      ans += inputString[j * n + i];
    }
  }

  return ans;
}
